components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/cache/PolicyCache.java

```diff
@@ -78,6 +78,11 @@ public static void updateLocalPolicyCacheMap(IdentityCacheKey identityCacheKey,
                 synchronized (localPolicyCacheMap) {
                     if (localPolicyCacheMap.get(identityCacheKey.getTenantId()) != null) {
                         if(localPolicyCacheMap.get(identityCacheKey.getTenantId()).get(identityCacheKey.getKey())!=null){
+                            if (policyStatus != null && policyStatus.getPolicyId()
+                                    .equals(localPolicyCacheMap.get(identityCacheKey.getTenantId())
+                                            .get(identityCacheKey.getKey()).getPolicyId())) {
```

Comment:
Can `policyStatus.getPolicyId` or `get(identityCacheKey.getKey())` return null?

Reply:
From here https://github.com/wso2/carbon-identity-framework/pull/3411/files#diff-04891aa75f334e88a5beea15344f296c21ac752cc6f7dbce3122b55dc96aa344R214, if the policyStatus is null, it will create the policyStatus with id, and the null check for `get(identityCacheKey.getKey())` is already handled from  https://github.com/wso2/carbon-identity-framework/pull/3411/files#diff-04891aa75f334e88a5beea15344f296c21ac752cc6f7dbce3122b55dc96aa344R73